pyteal/ir/tealsimpleblock.py

```diff
@@ -24,16 +26,35 @@ def replaceOutgoing(self, oldBlock: TealBlock, newBlock: TealBlock) -> None:
             self.nextBlock = newBlock
     
     def __repr__(self) -> str:
-        return "TealSimpleBlock({}, next={})".format(
+        # check for loop
+        if self in self.visited:
+            return "TealSimpleBlock({}, next={})".format(
+                repr(self.ops),
+                "",
+            )
+        self.visited.append(self)
+
+        s = "TealSimpleBlock({}, next={})".format(
             repr(self.ops),
             repr(self.nextBlock),
         )
+
+        self.visited.pop()
+        return s
     
     def __eq__(self, other: object) -> bool:
+        # check for loop
+        if self in self.blocks:
```

Comment:
I don't love that these functions need to modify the object in order to work, but I can't think of a better solution right now -- I might make a follow up PR if I can think of something better.

In any case, `self.blocks` and `self.visited` do not need to be lists, since they only ever contain `self` or nothing. A cleaner solution would be to make a single `self.visited` boolean attribute, and the functions can set this to true/false to indicate whether `self` has been visited.